### PR TITLE
#90 - Obtains `index.yaml` by directory

### PR DESCRIPTION
**Pull request by genryxy**
Part of #90 
Implemented obtaining `index.yaml` file by directory. In next PR I'll add some negative test cases.

### CODE REVIEW

File: src/main/java/com/artipie/helm/metadata/IndexByDirectory.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                    .thenCompose(PublisherAs::bytes)
+                    .thenApply(TgzArchive::new)
+                    .thenApply(TgzArchive::chartYaml)
+                    .thenAcceptBoth(
+                        this.index(),
+                        (chart, index) -> IndexByDirectory.addEntry(chart, index, entries)
+                    )
+                ).map(CompletableFuture::join)
+                .map(CompletableFuture::completedFuture)
+                .toArray(CompletableFuture[]::new)
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/helm/metadata/IndexByDirectory.java:92]: @genryxy hm, I'm not sure operations `join` and `completedFuture` are required here, seems like we can collect futures `toArray` right after `thenAcceptBoth`. Moreover, calling blocking `join` here is not a good idea

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@genryxy thanks, please, check my comment

**Comment by genryxy** [comment]
@olenagerasimova thanks, sometimes there were problems with concurrency (approximately 10 out of 100 attempts failed) therefore I added `join`. I wrongly assumed that there would be a non-blocking call of `join` as it is nested. Now it was corrected and all test's runs out of 1000 were successful.

**Review by olenagerasimova — APPROVED** [review]
@genryxy thanks, looks good to me

**Comment by 0crat** [comment]
Job `gh:artipie/helm-adapter#91` is not assigned, can't get performer

**Pull request merged**